prusti-contracts/prusti-specs/src/extern_spec_rewriter/traits.rs

```diff
@@ -49,29 +59,42 @@ fn generate_new_struct(item_trait: &syn::ItemTrait) -> syn::Result<GeneratedStru
 
     // Add a new type parameter to struct which represents an implementation of the trait
     let self_type_ident = syn::Ident::new("Prusti_T_Self", item_trait.span());
-    new_struct.generics.params.push(syn::GenericParam::Type(
-        parse_quote!(#self_type_ident),
-    ));
+    new_struct
+        .generics
+        .params
+        .push(syn::GenericParam::Type(parse_quote!(#self_type_ident)));
 
     let parsed_generics = parse_trait_type_params(item_trait)?;
     // Generic type parameters are added as generics to the struct
     for parsed_generic in parsed_generics.iter() {
         if let ProvidedTypeParam::GenericType(type_param) = parsed_generic {
-            new_struct.generics.params.push(syn::GenericParam::Type(type_param.clone()));
+            new_struct
+                .generics
+                .params
+                .push(syn::GenericParam::Type(type_param.clone()));
         }
     }
 
     let self_type_trait: syn::TypePath = parse_quote_spanned! {item_trait.span()=>
-        #trait_ident :: <#(#parsed_generics),*>
+        #trait_path :: <#(#parsed_generics),*>
     };
 
     // Add a where clause which restricts this self type parameter to the trait
-    if let Some(where_clause) = &item_trait.generics.where_clause {
-        return Err(syn::Error::new(where_clause.span(), "Where clauses for extern traits specs are not supported"));
-    }
-    let self_where_clause: syn::WhereClause = parse_quote! {
-        where #self_type_ident: #self_type_trait
-    };
+    let self_where_clause: syn::WhereClause =
```

Comment:
Maybe worht checking: is a `trait Foo where Self: Bar ...` clause handled correctly, i.e. is the `Self` rewritten to `Prusti_T_Self` as well?

Reply:
Good catch! It indeed does not. Funnily enough it appears it would still work, if not for the fact that `Self` does not appear to be getting rewritten in pre-/postconditions anyway, so this area might need some more love in general. I'll handle the where clauses though.

Reply:
On second thought, I'm not sure what I needed where clauses in extern specs for anymore? I don't have any code lying around using it, and it doesn't seem useful, since you don't need them to figure out which trait it is, and it seems you already get the type bounds for "free", i.e. you're able to use methods from `Bar` in the specs for `Foo` like this:

```rs
#[extern_spec]
trait Foo {
    #[ensures(result == self.bar())]
    fn foo(&self) -> i32;
}
```

(I initially wrote this as static methods rather than taking `&self`, bringing the `Self`-in-specs issue to my attention.)

Reply:
Ah right, you're right: the bounds on `Self` should of course be inherited from the actual trait definition. Never mind!